### PR TITLE
Add TriBITS compatibility tests

### PR DESCRIPTION
**Pull request by dalg24**
As it says on the tin

### CODE REVIEW

File: core/unit_test/CMakeLists.txt [hunk 1]
```diff
@@ -1284,6 +1284,29 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
...
+KOKKOS_ADD_TEST(
+  NAME CoreUnitTest_CMakeTriBITSCompatibilityExtraArgs
+  EXE CoreUnitTest_CMakeTriBITSCompatibility
+)
+
+SET(Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityEnvironment_ENVIRONMENT "KOKKOS_TEST_TRIBITS_COMPATIBILITY=1")
+KOKKOS_ADD_TEST(
+  NAME CoreUnitTest_CMakeTriBITSCompatibilityEnvironment
+  EXE CoreUnitTest_CMakeTriBITSCompatibility
+)
```
Comment (masterleinad) [inline core/unit_test/CMakeLists.txt:1308]: Setting `ENVIRONMENT` via a CMake variable is not supported by `TriBITS` but only by `tribits_add_test` directly. Do we want to keep it? I now lean slightly to removing until there is demand.

Also, we could add `<fullTestName>_SET_DISABLED_AND_MSG` for better compatibility with `TriBITS`.

Comment (dalg24) [reply core/unit_test/CMakeLists.txt:1308]: I know it is not a TriBITS option but could not think of a better way to categorize it.
I could add a comment if you want me to.

As far as _SET_DISABLED_AND_MSG is concerned, I was not planning to tackle it here.

File: core/unit_test/CMakeLists.txt [hunk 2]
```diff
@@ -1284,6 +1284,29 @@ KOKKOS_ADD_EXECUTABLE_AND_TEST(
   ARGS "one 2 THREE"
 )
 
+KOKKOS_ADD_EXECUTABLE(
+  CoreUnitTest_CMakeTriBITSCompatibility
+  SOURCES UnitTest_CMakeTriBITSCompatibility.cpp
+)
+
+SET(Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityDisable_DISABLE ON)
```
Comment (ndellingwood) [inline core/unit_test/CMakeLists.txt:1292]: To check my understanding, the confirmation that this test passes (i.e. does not run thanks to the _DISABLE property) is that the unit test will not terminate with `EXIT_FAILURE`, even though the test lacks the required  `KOKKOS_TEST_TRIBITS_COMPATIBILITY` or `--kokkos-test-tribits-compatibility` matches required for success, does that sound correct?

**Review by ndellingwood — APPROVED** [review]
Nice tests, thanks for adding

File: core/unit_test/CMakeLists.txt (hunk 2 above)
Comment (ndellingwood) [reply core/unit_test/CMakeLists.txt:1292]: I'm referring to L1292-1296, my highlight block didn't carry over when I commented

Comment (dalg24) [reply core/unit_test/CMakeLists.txt:1292]: Correct, if we actually tried to run that command it would fail.

Comment (dalg24) [reply core/unit_test/CMakeLists.txt:1298]: I added a test with the WILL_FAIL property to ensure that it would indeed fail

Comment (dalg24) [reply core/unit_test/CMakeLists.txt:1298]:
```
$ ctest --test-dir cmake-build-default -C Debug -R TriBITS

Internal ctest changing into directory: /Users/qdi/CLionProjects/kokkos/cmake-build-default
Test project /Users/qdi/CLionProjects/kokkos/cmake-build-default
    Start 19: Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityWillFail
1/3 Test #19: Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityWillFail ......   Passed    0.62 sec
    Start 20: Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityExtraArgs
2/3 Test #20: Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityExtraArgs .....   Passed    0.02 sec
    Start 21: Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityEnvironment
3/3 Test #21: Kokkos_CoreUnitTest_CMakeTriBITSCompatibilityEnvironment ...   Passed    0.01 sec

100% tests passed, 0 tests failed out of 3

Label Time Summary:
Kokkos    =   0.64 sec*proc (3 tests)

Total Test time (real) =   0.66 sec

**Comment by dalg24** [comment]
Failure was unrelated (GPU not detected at runtime in one of the CUDA builds)

**Review by crtrott — APPROVED** [review]

**Pull request merged**